ral-facilities/operationsgateway
# DSEGOG-65 DSEGOG-150 use all records in plotting, choose channels from record table

## Description
Addresses two issues:
- Plots can now use all data available from the records endpoint. Previously, it was only using the paginated data from the record table. 
- Users can now choose scalar channels to plot that are currently present in the record table. This is done via a `Select` above the existing `Autocomplete` which is used to choose from any available channels. 

A significant amount of refactoring has been done to harmonise the channel metadata available to all components which want them. The Redux store previously contained the availableColumns, which only contained information about the channel name. This wasn't useful for the plot to determine which table columns contained scalar data that could be plotted. Therefore, this record of availableColumns has been replaced with an array of FullChannelMetadata, which is more useful for the plot and potentially more useful for future functionality involving data channel searching, viewing images and traces, etc. The only place where availableColumns is still used is in the recordTable, which is the only place that realistically needs that information.

The potential downside to this is that the shot number, active area and active experiment columns are now also treated as data channels to accommodate this. This may not make a lot of sense but was easiest when doing this, although we'd have to still treat those "channels" as special cases when the user is wanting to browse or search channels. I'm open to suggestions on whether this is a good idea and if it should be changed.

## Testing instructions
Populate the record table with a mixture of scalar and non-scalar columns and then try and plot them while making sure only the scalar channels are selectable. Also check that any data that is plotted is not from a paginated query, i.e. it contains all data we have on this channel.


## Agile board tracking
Closes DSEGOG-65 DSEGOG-150
